src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -607,6 +607,15 @@ private Expression<?> parseSingleExpr(boolean allowUnparsedLiteral, @Nullable Lo
 					return new SimpleLiteral<>(parsedObject, false, new UnparsedLiteral(expr));
 				}
 			}
+			if (expr.startsWith("\"") && expr.endsWith("\"")) {
+				for (ClassInfo<?> aClass : exprInfo.classes) {
+					if (!aClass.getC().isAssignableFrom(String.class))
+						continue;
+					VariableString string = VariableString.newInstance(expr.substring(1, expr.length() - 1));
+					if (string instanceof LiteralString)
+						return string;
```

Comment:
```suggestion
						return string;
					break;
```
I'm thinking we should break if it is not a valid variablestring (the loop has no impact on whether it is valid)